src/utils/url.ts

```diff
@@ -56,3 +61,170 @@ export function addUrlQuery(url: string, query: Record<string, unknown>): string
+  }
+  /* istanbul ignore next */
+  if (typeof dataKey !== "string") {
+    throw new Error(`Expected parameter 'dataKey' to be type string, was type ${typeof dataKey}`);
+  }
+
+  const opts = {
+    ...defaultGetEntryOptions,
+    ...customOptions,
+  };
+
+  const timeout = opts.timeout;
+
+  if (!Number.isInteger(timeout) || timeout > MAX_GET_ENTRY_TIMEOUT || timeout < 1) {
+    throw new Error(
+      `Invalid 'timeout' parameter '${timeout}', needs to be an integer between 1s and ${MAX_GET_ENTRY_TIMEOUT}s`
+    );
+  }
+
+  // Trim the prefix if it was passed in.
+  publicKey = trimPrefix(publicKey, "ed25519:");
+  if (!isHexString(publicKey)) {
+    throw new Error(`Given public key '${publicKey}' is not a valid hex-encoded string or contains an invalid prefix`);
+  }
+
+  const query = {
+    publickey: `ed25519:${publicKey}`,
+    datakey: toHexString(hashDataKey(dataKey)),
+    timeout,
+  };
+
+  let url = makeUrl(portalUrl, opts.endpointPath);
+  url = addUrlQuery(url, query);
+
+  return url;
+}
+
+/**
+ * Gets the skylink URL without an initialized client.
+ *
+ * @param portalUrl - The portal URL.
+ * @param skylinkUrl - Skylink string. See `downloadFile`.
+ * @param [customOptions] - Additional settings that can optionally be set.
+ * @returns - The full URL for the skylink.
+ * @throws - Will throw if the skylinkUrl does not contain a skylink or if the path option is not a string.
+ */
+export function getSkylinkUrlForPortal(
+  portalUrl: string,
+  skylinkUrl: string,
+  customOptions?: CustomDownloadOptions
+): string {
+  /* istanbul ignore next */
+  if (typeof portalUrl !== "string") {
+    throw new Error(`Expected parameter 'portalUrl' to be type string, was type ${typeof portalUrl}`);
+  }
+  /* istanbul ignore next */
+  if (typeof skylinkUrl !== "string") {
+    throw new Error(`Expected parameter skylinkUrl to be type string, was type ${typeof skylinkUrl}`);
+  }
+
+  const opts = { ...defaultDownloadOptions, ...customOptions };
+
+  const query = opts.query ?? {};
+  if (opts.download) {
+    // Set the "attachment" parameter.
+    query.attachment = true;
+  }
+  if (opts.noResponseMetadata) {
+    // Set the "no-response-metadata" parameter.
+    query["no-response-metadata"] = true;
+  }
+
+  // URL-encode the path.
+  let path = "";
+  if (opts.path) {
+    if (typeof opts.path !== "string") {
+      throw new Error(`opts.path has to be a string, ${typeof opts.path} provided`);
+    }
+
+    // Encode each element of the path separately and join them.
+    //
+    // Don't use encodeURI because it does not encode characters such as '?'
+    // etc. These are allowed as filenames on Skynet and should be encoded so
+    // they are not treated as URL separators.
+    path = opts.path
+      .split("/")
+      .map((element: string) => encodeURIComponent(element))
+      .join("/");
+  }
+
+  let url;
+  if (opts.subdomain) {
```

Comment:
I don't understand this conditional. Can we drop a comment explaining it a little? It seems `opts.subdomain` is not used in any of the blocks?

Reply:
It's a boolean. It's not part of this PR, feel free to write an issue for later.

Reply:
https://github.com/SkynetHQ/skynet-js/issues/26